### PR TITLE
Pix: Fix crash getting type name of unnamed structs

### PR DESCRIPTION
**Pull request by jeffnn**
...and other types while I'm at it. There are tests for all of the possible unnamed types (structs, members, arrays). Don't think I can express an unnamed scalar or typededf so there's no test for those.
(This was hit in the wild while debugging a pixel shader in a WinPIX capture of a Large Well-Known Engine.)

### CODE REVIEW

**Review by tex3d — APPROVED** [review]
LGTM!

File: lib/DxilDia/DxcPixTypes.cpp [hunk 1]
```diff
@@ -14,6 +14,13 @@
 #include "DxcPixTypes.h"
 #include "DxilDiaSession.h"
 
+static const char * GetTypeNameOrDefault(llvm::DIType *diType) {
```
Comment (pow2clk) [inline lib/DxilDia/DxcPixTypes.cpp:17]: Given that every use of this converts the output from ascii to widechar, perhaps this should return widechar instead?

**Review by pow2clk — APPROVED** [review]
Looks good. I just have one potential cleanup.

I think it would be clearer if you spent some time in the description explaining what you do test instead of just what you don't. I went into this thinking there were no tests at all!

File: lib/DxilDia/DxcPixTypes.cpp (hunk 1 above)
Comment (jeffnn) [reply lib/DxilDia/DxcPixTypes.cpp:17]: Good call. Thanks.

**Pull request merged**